src/Rack.php

```diff
@@ -708,6 +711,9 @@ public function prepareInputForAdd($input)
 
     public function prepareInputForUpdate($input)
     {
+        if (!isset($input['bgcolor']) || empty($input['bgcolor'])) {
```

Comment:
When update is made from something else than the UI form (e.g. API call), this field may be voluntarly not passed. As its value is forced when rack is created, we should only verify that its values is not erased by an empty value.

```suggestion
        if (array_key_exists('bgcolor', $input) && empty($input['bgcolor'])) {
```